### ISSUE TITLE
Change Subnet IP

### ISSUE THREAD

**Issue by KBapna** [hwdsl2/setup-ipsec-vpn#1304]
**Checklist**

- [ x] I searched existing [Issues](https://github.com/hwdsl2/setup-ipsec-vpn/issues?q=is%3Aissue), and did not find a similar enhancement request
- [ x] This enhancement request is about the VPN setup scripts, and not IPsec VPN itself
- [x ] I read the [README](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/README.md)
- [ x] I read the [Important notes](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/README.md#important-notes)
- [ x] I followed instructions to [configure VPN clients](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/README.md#next-steps)
- [ x] I checked [Troubleshooting](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients.md#troubleshooting) and [VPN status](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients.md#check-logs-and-vpn-status)

**Discription**

I was trying to install VPN on Proxmox Server VE as a VM (not CT) and wanted its users to be part of subnet 192.168.2.x and communicate 2 way (transfer files as if they were physically on subnet ignore speed problems) but

1) I cant seem to change the prefix (192.168.2. ) even after changing it in vpn.sh.
2) Cant get VPN users or server to ping each other ( 192.168.2.1 is gateway witch DHCP and internet)

**Enhancement Request**

Add a method or command to change prefix ip ?

**Additional Info**

Proxmox is on subnet 192.168.0.x and VPN server is on 192.168.0.3 with its gateway being 192.168.0.1

**Comment by hwdsl2** [comment]
@KBapna Hello! To customize VPN subnet, please refer to:
https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/advanced-usage.md#customize-vpn-subnets

As mentioned in that section, if you have already set up the VPN, you may need to first uninstall it, then specify custom subnets.

Client-to-client traffic is enabled by default. For communication among VPN clients and the VPN server, refer to the following sections for more information. Also make sure that the VPN clients' firewalls allow the desired traffic for your use case.
https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/advanced-usage.md#internal-vpn-ips-and-traffic
https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/advanced-usage.md#access-vpn-servers-subnet
https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/advanced-usage.md#access-vpn-clients-from-servers-subnet

**Comment by KBapna** [comment]
Hi, Thank You for the quick reply
Sorry I wasn't able to find these guides earlier but I have ran into another problem.

Current setup :
VPN server IP, Gateway, eth       :     192.168.0.3/24, 192.168.0.1  eth0
VPN client range                        :     192.168.2.1-192.168.2.250
VM1 IP, Gateway, eth                 :     192.168.0.2/24  192.168.0.1  eth0
VM2 IP, Gateway, eth                 :     192.168.1.2/24  192.168.1.1  eth0

All VPN client are able to connect to VPN server and subnet as expected
But VMs are not able to connect back even after setting up IP rules on VPN server and OPNsense

OPNsense config
Static Route:   
      Network: 192.168.2.0/24
      Gateway: 192.168.0.3

VPN Iptables:
For 192.168.0.x subnet:
netif=$(route 2>/dev/null | grep -m 1 '^default' | grep -o '[^ ]*$')
iptables -I FORWARD 2 -i "$netif" -o ppp+ -s 192.168.0.0/24 -j ACCEPT
iptables -I FORWARD 2 -i "$netif" -d 192.168.2.0/24 -s 192.168.0.0/24 -j ACCEPT

and I have put them to rc.local for auto apply

Also have noticed that If I connect VM3 to VPN server and give it IP 192.168.2.10, everyone can ping it but if I give it 192.168.1.10 no one can ping it as expected from static rules.

So, If server can ping clients, that might do the trick.

**Comment by hwdsl2** [comment]
@KBapna Yes, I think you can try changing 192.168.0.0/24 to 192.168.0.0/22 for your use case, and see if it works. The static rules should be added to the system you want to access VPN clients from (if not already done so). If you want to troubleshoot further, one method is to temporarily remove the last DROP rule from the IPTables FORWARD chain, and add a LOG rule instead. Add the DROP rule back when finished.

**Comment by KBapna** [comment]
Hey,
I changed it from 192.168.0.0/24 to 22 and also started logging but logs are empty even when I ping each other.
I also noticed that when from VM1 I ping 192.168.2.1 (Not existing on network at all) I get reply

PING 192.168.2.1 (192.168.2.1) 56(84) bytes of data.
From 192.168.0.3 icmp_seq=1 Time to live exceeded
From 192.168.0.3 icmp_seq=2 Time to live exceeded
From 192.168.0.3 icmp_seq=3 Time to live exceeded
From 192.168.0.3 icmp_seq=4 Time to live exceeded
From 192.168.0.3 icmp_seq=5 Time to live exceeded

so looks like req is reaching the VPN server but TTL as there is no client as expected.
Now when I add client to 192.168.2.2 and ping I get no response. Like completely blank, looks like a firewall problem.
All I have done to allow traffic from VPN server on windows is to add mar it is "private network" and looks like traffic is still  being blocked.
